web/models/attendance/attendance_helper.ex

```diff
@@ -84,4 +84,110 @@ defmodule CoursePlanner.AttendanceHelper do
 
     Repo.transaction(updated_multi)
   end
+
+  def update_class_attendance_records(offered_course_id) do
+    query = from oc in OfferedCourse,
+    join: s in assoc(oc, :students),
+    join: c in assoc(oc, :classes),
+    join: cs in assoc(c, :students),
+    preload: [students: s, classes: {c, students: cs}]
+
+    offered_course_with_classes     = Repo.get(query, offered_course_id)
+    offered_course_with_attendances = get_course_attendances(offered_course_id)
+
+    cond do
+      is_nil(offered_course_with_attendances) && is_nil(offered_course_with_classes) ->
+        {:ok, nil}
+      is_nil(offered_course_with_attendances) ->
+        Enum.map(offered_course_with_attendances.classes, fn(class) ->
+          create_class_attendance_records(class)
+        end)
+      true ->
+        offered_course_with_attendances
+        |> get_incorrect_attendance_record()
+        |> fix_incorrect_attendance_record()
+    end
+  end
+
+  defp get_incorrect_attendance_record(offered_course) do
+    offered_course_classes = offered_course.classes
+
+    Enum.map(offered_course_classes, fn(class) ->
+      missing_students =
+        Enum.filter(offered_course.students, fn(student) ->
+          not Enum.any?(class.attendances, fn(attendance) ->
+                attendance.student.id == student.id
+              end)
+        end)
+
+        excessive_attendances =
+          Enum.filter(class.attendances, fn(attendance) ->
+            not Enum.any?(offered_course.students, fn(student) ->
+                  attendance.student.id == student.id
+                end)
+          end)
+
+        %{
+          class_id: class.id,
+          missing_students: missing_students,
+          excessive_attendances: excessive_attendances
+         }
+    end)
+  end
+
+  defp fix_incorrect_attendance_record(incorrect_records)
+    when is_list(incorrect_records) do
```

Comment:
Do you have a case where this is not a `list`? You should either remove this condition or add another call returning an error when it's not.